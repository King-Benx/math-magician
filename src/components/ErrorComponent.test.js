import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ErrorComponent from './ErrorComponent';

test('The error component renders correctly', () => {
  render(<ErrorComponent error loading={false} />);
  const sampleContent = screen.getByText('Something went wrong and we failed to get the quote');
  expect(sampleContent).toBeTruthy();
  expect(sampleContent).toMatchSnapshot();
});

test('the error component renders fully', () => {
  const tree = renderer.create(<ErrorComponent error loading={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
