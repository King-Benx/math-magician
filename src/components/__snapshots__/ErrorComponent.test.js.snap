// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The error component renders correctly 1`] = `
<span
  class="error"
>
  Something went wrong and we failed to get the quote
</span>
`;

exports[`the error component renders fully 1`] = `
<div
  className="quote"
>
  <span
    className="error"
  >
    Something went wrong and we failed to get the quote
  </span>
</div>
`;
